File: Cantidad de dígitos de un número/Python/cuenta_digitos_v3.py
#Creado por Andrés Melgar
#para iterando++ http://iterando.online/
#entrada: http://iterando.online/ejercicios-resueltos/cantidad-de-digitos-de-un-numero-en-python/
import math

def cuenta_digitos(numero):
    if numero == 0:
        return 1
    return int(math.log10(numero) + 1)

numero=int(input("Ingrese un número: "))
numero=abs(numero)
print("El número tiene {:d} dígito(s)".format(cuenta_digitos(numero)))

File: Cantidad de dígitos de un número/Java/Cuenta_Digitos_v3.java
package cuenta_digitos_v3;

import static java.lang.Math.abs;
import java.util.Scanner;

public class Cuenta_Digitos_v3 {

    public static void main(String[] args) {
        int numero;
        Scanner reader = new Scanner(System.in);
        System.out.printf("Ingrese un número: ");
        numero = reader.nextInt();

        numero = abs(numero);
        System.out.printf("El número tiene %d dígito(s)%n", cuenta_digitos(numero));
    }

    private static int cuenta_digitos(int numero) {
        int total_digitos = 0;
        do {
            total_digitos++;
            numero /= 10;
        } while (numero > 0);
        return total_digitos;
    }

}
